### ISSUE TITLE
Error handling when trust_bundle_file is not specified

### ISSUE THREAD

**Issue by jmg421** [Venafi/vault-pki-backend-venafi#79]
**PROBLEM SUMMARY**
A generic error message may result when `trust_bundle_file` is not specified during a `vault write -address https://vault.amplisight.io:8200 venafi-pki/venafi/tpp url="https://tpp-amplisight-trial.venafidemo.com/vedsdk" access_token="XXX" refresh_token="XXX" zone="Vault-DevSecOps-Demo"` command is run.

The resulting error is:
```
Error writing data to venafi-pki/venafi/tpp: Error making API request.
URL: PUT https://vault.amplisight.io:8200/v1/venafi-pki/venafi/tpp
Code: 500. Errors:
* 1 error occurred:
        * internal error

```
**EXPECTED RESULTS**
The expected result is `Success! Data written to: venafi-pki/venafi/tpp`

**COMMENTS/WORKAROUNDS**
A better error message might direct the user to specify the `trust_bundle_file`

**Comment by tr1ck3r** [comment]
Thank you @jmg421.  We'll investigate whether we can improve the error messaging but we're somewhat at the mercy of what the Go http client returns.  The underlying problem was that the `trust_bundle_file` parameter was required when the certificate securing the TPP API was issued by a CA that is trusted at the operating system level.  The expected behavior is that the `trust_bundle_file` should only be required when the certificate securing the TPP API was issued by a CA that is _NOT_ trusted by the OS.  The TPP API is commonly secured using a non-publicly trusted (enterprise PKI) certificate but there's no reason it can't be secured by a publicly trusted certificate and customers should have the option to manage trust at the operating system level as opposed to the integration level.

**Comment by tr1ck3r** [comment]
I'm able to reproduce this in my environment but the error is slightly different:
```
Error writing data to pki-backend/venafi/tpp: Error making API request.

URL: PUT https://vault.venafi.example/v1/pki-backend/venafi/tpp
Code: 500. Errors:

* 1 error occurred:
        * rpc error: code = Unavailable desc = transport is closing
```
My system has the trust anchor (root CA) certificate applicable to my TPP API in the operating system trust store.

**Comment by jmg421** [comment]
@tr1ck3r yes we saw that variety of the error message as well.

**Comment by phishbandana** [comment]
I am experiencing a similar error. So the question - is there a work around? do we need to include the trust_bundle_file? 

when running the similar write request without the trust_bundle_file the output is still internal error.

```
vault write pki/venafi/tpp url="https://URI-goes-here/VEDSDK" \
access_token="...." \
refresh_token="...." \
zone=".../..."

```
the output is

```
Error writing data to pki/venafi/tpp: Error making API request.

URL: PUT https://vaultaddress:8200/v1/pki/venafi/tpp
Code: 500. Errors:

* 1 error occurred:
	* internal error
```

When doing the same write but with the trust_bundle_file, but with out the zone details, the output is the same. Further, the vault documentation is: 
Trust Protection Platform:
[Vault Link to Docs](https://www.vaultproject.io/docs/secrets/venafi)

```
$ vault write venafi-pki/venafi/tpp \
    url="https://tpp.venafi.example" \
    access_token="tn1PwE1QTZorXmvnTowSyA==" \
    refresh_token="MGxV7DzNnclQi9CkJMCXCg==" \
    zone="DevOps\\HashiCorp Vault" \
    trust_bundle_file="/path-to/bundle.pem"
Success! Data written to: venafi-pki/venafi/tpp
```

Venafi docs
state:
Trust Protection Platform:

```
$ vault write pki/venafi/tpp \
    url="https://tpp.example.com" trust_bundle_file="/path/to/bundle.pem" \
    access_token="tn1PwE1QTZorXmvnTowSyA==" refresh_token="MGxV7DzNnclQi9CkJMCXCg=="

```
Which one is the correct one that should be used?

thanks!

**Comment by tr1ck3r** [comment]
@phishbandana yes, the workaround is to specify the `trust_bundle_file` parameter referencing a text file that contains the root CA certificate of the trust chain that issued the TPP API server certificate.

Both of the docs you references are actually correct, it just depends on how you choose to set things up.  You have the option to specify the `zone` on the Venafi secret or the PKI role.  So for the second doc reference the `zone` must be specified on the PKI role since it is not being specified on the Venafi secret.  If you specify the `zone` in both places, the `zone` value of the PKI role applies.

**Comment by phishbandana** [comment]
Hey Ryan. Thank you for the reply. As we understand it, we have the root Cert in our file location. Does the location of the file matter? That is; we are running on GKE and have three vault pods. Since the opt files are owned by the root user, I have placed (for POC) the files in another directory.

I will try again with different cert and report back. Thanks again.

Thank you,
Alan

**Comment by angelmoo** [comment]
Hi @phishbandana the location of the file shouldn't matter, you will need to specify the absolute path of the certificate on the trust_bundle_file property

**Comment by tr1ck3r** [comment]
This issue has been resolved in the [v0.8.3](https://github.com/Venafi/vault-pki-backend-venafi/releases/tag/v0.8.3) release.
